### ISSUE TITLE
Gitup no longer works with GitHub

### ISSUE THREAD

**Issue by fushi** [git-up/GitUp#421]
This is likely being caused by https://githubengineering.com/crypto-removal-notice/ (per #241).

**Comment by hawkrives** [comment]
It still works for me (on High Sierra)?

**Comment by combs** [comment]
Completely dead in the water for me

**Comment by gabehoffman** [comment]
not working at all for me

**Comment by luongvo209** [comment]
not working for me (High Sierra, gitup 1.0.9). But guys , I think the latest release will fix this problem for us. Waiting for an update to come 😄  
![image](https://user-images.githubusercontent.com/15828926/36576250-0cc549e2-1882-11e8-849c-de0a36dc9c81.png)

**Comment by swisspol** [comment]
Can anyone try with 1.0.10?

**Comment by swisspol** [comment]
Looks an updated version of libssh2 is needed

**Comment by meduzen** [comment]
I tried with GitUp 1.0.9 and 1.0.10, the error remains the same no matter the GitHub project:
> Failed connecting to "origin" remote: Failed to start SSH session: Unable to exchange encryption keys

Works on other repositories servers (private, or gitlab.com). And MacOS `git something` works on github.com.

MacOS 10.12.6 here. Will also try this week-end on a MacOS 10.13.xx computer and let you know (if no fix comes meanwhile).

**Comment by filiptepper** [comment]
No luck, GitUp 1.0.10, macOS 10.13.3.

**Comment by mluisbrown** [comment]
I built a version on my local machine using version `1.8.0` of libssh2 (as mentioned [here](https://github.com/git-up/GitUp/issues/241#issuecomment-367890389)) and that version works fine.

Just a question of using an updated libssh2 which supports the newer crypto standards now enforced by GitHub.

**Comment by Gryzor** [comment]
I also tried w/ 1.0.10 (and reverted to 9), high sierra. Same problem.

**Comment by mluisbrown** [comment]
@Gryzor the problem is independent of the version of GitUp and of the version of the OS. All current GitUp versions use the same old version of `libssh2` which uses standards that GitHub no longer support. Until there is an updated version of GitUp using a more recent version of `libssh2`, this problem will continue.

**Comment by Gryzor** [comment]
@mluisbrown I gathered that from the thread, thanks. Just added the info because it appeared @swisspol didn’t know if this was the case; I commented from mobile, so I didn’t have the full thread reloaded. 👍

**Comment by csjones** [comment]
Started getting this error yesterday but was able to resolve the issue by changing `remote.origin.url` in the git config from `git@github.com:csjones/gg.git` to `https://github.com/csjones/gg`.

**Comment by fushi** [comment]
@csjones Yes, using HTTPS instead of SSH fixes the problem.

**Comment by meduzen** [comment]
Yes, it _fixes_ the problem, but then you have to provide credentials on each interaction with your GitHub repository, unless you give GitUp full access to your Keychain.

**Comment by douglashill** [comment]
I put up a build of my fork with libssh2 1.8.0 as described above: https://github.com/douglashill/GitUp/releases

**Comment by luongvo209** [comment]
@douglashill Can you open a pull request for upstream to fix the problem?

**Comment by douglashill** [comment]
I’m not going to submit a pull request, but if someone else wants to take my changes and do that go ahead.

**Comment by mluisbrown** [comment]
I have created a PR for `Libraries` to update the version of `libssh2`. Once that is approved, a new PR can be created to point to the new version of the `Libraries` submodule.

**Comment by swisspol** [comment]
Should hopefully be fixed now in master.
